activiti-cloud-services-runtime-bundle/activiti-cloud-services-rest-impl/src/main/java/org/activiti/cloud/services/rest/controllers/ProcessVariablesPayloadValidator.java

```diff
@@ -26,102 +26,102 @@
 import org.activiti.api.process.model.payloads.StartProcessPayload;
 import org.activiti.common.util.DateFormatterProvider;
 import org.activiti.engine.ActivitiException;
+import org.activiti.spring.process.ProcessExtensionService;
 import org.activiti.spring.process.model.Extension;
 import org.activiti.spring.process.model.ProcessExtensionModel;
 import org.activiti.spring.process.model.VariableDefinition;
 import org.activiti.spring.process.variable.VariableValidationService;
 
-public class ProcessVariablesPayloadValidator  {
-   
-    private final Map<String, ProcessExtensionModel> processExtensionModelMap;
+public class ProcessVariablesPayloadValidator {
+
     private final VariableValidationService variableValidationService;
     private final DateFormatterProvider dateFormatterProvider;
- 
+    private ProcessExtensionService processExtensionService;
+
     public ProcessVariablesPayloadValidator(DateFormatterProvider dateFormatterProvider,
-                                            Map<String, ProcessExtensionModel> processExtensionModelMap,
+                                            ProcessExtensionService processExtensionService,
                                             VariableValidationService variableValidationService) {
         this.dateFormatterProvider = dateFormatterProvider;
-        this.processExtensionModelMap = processExtensionModelMap;
+        this.processExtensionService = processExtensionService;
         this.variableValidationService = variableValidationService;
     }
-    
+
     private Optional<Map<String, VariableDefinition>> getVariableDefinitionMap(String processDefinitionKey) {
-        ProcessExtensionModel processExtensionModel = processExtensionModelMap.get(processDefinitionKey);
+        ProcessExtensionModel processExtensionModel = processExtensionService.getExtensionsForId(processDefinitionKey);
```

Comment:
Here process extension service is expecting `processDefinitionId`, but you're passing `processDefinitionKey`. I think you can rename the parameter to `processDefinitionId` and make sure that all the callers are passing `id` and not `key`. For instance, https://github.com/Activiti/activiti-cloud-runtime-bundle-service/blob/develop/activiti-cloud-services-runtime-bundle/activiti-cloud-services-rest-impl/src/main/java/org/activiti/cloud/services/rest/controllers/ProcessInstanceVariableControllerImpl.java#L69